lib/engine/tile.rb

```diff
@@ -336,22 +337,25 @@ def compute_city_town_edges
         [ct, edge]
       end.to_h
 
-      # take care of city/towns with no paths
+      # take care of city/towns with no paths when there is one other city/town
       city_towns = @cities + @towns
       pathless_cts = city_towns.select { |ct| ct.paths.empty? }
       if pathless_cts.one? && city_towns.size == 2
         ct = pathless_cts.first
-        ct_edges[ct] = (ct_edges.values.first + 3) % 6
+        ct_edges[ct] = (ct_edges.values.first + 3) % 6 if ct_edges.values.first
       end
-      pathless_cts.select do |pct|
-        ct_edges[pct] = compute_loc(pct.loc) if pct.loc
+
+      # take care of city/towns with no exits
+      exitless_cts = city_towns.select { |xct| xct.exits.empty? }
+      exitless_cts.select do |xct|
+        ct_edges[xct] = compute_loc(xct.loc) if xct.loc
       end
 
       ct_edges
     end
 
     def revenue_to_render
-      @revenue_to_render ||= stops.map(&:revenue_to_render)
+      @revenue_to_render ||= (@cities + @towns + @offboards).map(&:revenue_to_render)
```

Comment:
why not stops?

Reply:
Only to handle the special case of pathless cities/towns. One could make a pretty decent argument that pathless revenue_centers shouldn't display revenue. Maybe that would be the better fix.

Reply:
i think we should fix the stops not working for pathless, that sounds wrong

Reply:
Well, technically is a stop without a path really a stop (from a routing standpoint)?